### PR TITLE
CharlesdevBookmark System Enhancement

### PR DESCRIPTION
**Pull request by Charles-Zhang-114514**
## Main Features Added
1. Folder Management System
   - Added support for custom bookmark folders
   - Implemented folder creation functionality
   - Added folder renaming capability

2. User Interface Improvements
   - Redesigned bookmark listing page to show folder structure
   - Added intuitive folder navigation
   - Maintained consistent UI style with existing design

3. Functionality Enhancements
   - Added folder-specific bookmark management
   - Implemented folder-level operations (clear, rename)
   - Maintained existing bookmark sharing functionality

## Technical Implementations
1. Updated Components:
   - BookmarkContext.tsx: Added folder management logic
   - BookmarkButton.tsx: Added folder selection modal
   - [folderId].tsx: New component for folder detail view
   - ResourcesHandouts.tsx: Enhanced to support folder-specific views

2. Data Management:
   - Extended localStorage implementation for folder support
   - Maintained existing bookmark data structure
   - Added new folder data structure
   - Implemented data consistency checks

## Testing
- Verified folder creation and management
- Tested bookmark organization across folders
- Confirmed sharing functionality works with folder system
- Validated UI responsiveness and consistency

Demo: https://github.com/user-attachments/assets/d894258e-cffd-4ce1-af3f-64327b98fe38

### CODE REVIEW

File: code/src/styles/Bookmark.module.css
Comment (mochiakku) [inline code/src/styles/Bookmark.module.css:1]: Could you remove the comments in chinese?

**Review by mochiakku — CHANGES_REQUESTED** [review]
Also we want to merge into the dev branch, not main, could you change that?

**Comment by mochiakku** [comment]
But besides that looks good! We definitely need to work on the styling but for now is good

**Comment by mochiakku** [comment]
Also, is there any way to create a new folder from the bookmark's page? Or are we only allowed to do that when we add a resource to a bookmark?

**Comment by Charles-Zhang-114514** [comment]
> Also, is there any way to create a new folder from the bookmark's page? Or are we only allowed to do that when we add a resource to a bookmark?

Hi, I will try tonight

**Comment by Charles-Zhang-114514** [comment]
> Also we want to merge into the dev branch, not main, could you change that?

OK, I have changed.

**Review by mochiakku — APPROVED** [review]

**Comment by mochiakku** [comment]
Looks good, be aware of the conflict before merging and I'll look into why the deployments are failing

**Comment by Charles-Zhang-114514** [comment]
> Looks good, be aware of the conflict before merging and I'll look into why the deployments are failing

Thank you, and if there is any problem, please let me know asap.

**Review by mochiakku — APPROVED** [review]

**Review by DarianC26 — APPROVED** [review]

**Pull request merged**